drivers/adc/adc_stm32.c

```diff
@@ -868,9 +869,9 @@ static void set_reg_value(const struct device *dev, uint32_t reg,
 	const struct adc_stm32_cfg *config = dev->config;
 	ADC_TypeDef *adc = config->base;
 
-	uintptr_t addr = (uintptr_t)adc + reg;
+	uint32_t *addr = (uint32_t *)adc + reg;
```

Comment:
To be fixed: the passed `reg` value is a byte offset. Here it becomes a 32bit word offset.
Suggestion:
```c
	const struct adc_stm32_cfg *config = dev->config;
	size_t reg32_offset = reg / sizeof(uint32_t);
	uint32_t *addr = (uint32_t *)config->base + reg32_offset;